routing/loaded_path_segment.hpp

```diff
@@ -30,29 +26,23 @@ using TSeg = OsrmMappingTypes::FtSeg;
 struct LoadedPathSegment
 {
   vector<m2::PointD> m_path;
+  vector<turns::SingleLaneInfo> m_lanes;
+  string m_name;
+  TEdgeWeight m_weight;
+  TNodeId m_nodeId;
   ftypes::HighwayClass m_highwayClass;
   bool m_onRoundabout;
   bool m_isLink;
-  TEdgeWeight m_weight;
-  string m_name;
-  TNodeId m_nodeId;
-  vector<SingleLaneInfo> m_lanes;
-
-  // General constructor.
-  LoadedPathSegment(RoutingMapping & mapping, Index const & index,
-                    RawPathData const & osrmPathSegment);
-  // Special constructor for side nodes. Splits OSRM node by information from the FeatureGraphNode.
-  LoadedPathSegment(RoutingMapping & mapping, Index const & index,
-                    RawPathData const & osrmPathSegment, FeatureGraphNode const & startGraphNode,
-                    FeatureGraphNode const & endGraphNode, bool isStartNode, bool isEndNode);
-
-private:
-  // Load information about road, that described as the sequence of FtSegs and start/end indexes in
-  // in it. For the side case, it has information about start/end graph nodes.
-  void LoadPathGeometry(buffer_vector<TSeg, 8> const & buffer, size_t startIndex,
-                        size_t endIndex, Index const & index, RoutingMapping & mapping,
-                        FeatureGraphNode const & startGraphNode,
-                        FeatureGraphNode const & endGraphNode, bool isStartNode, bool isEndNode);
+
+  LoadedPathSegment()
+    : m_weight(0)
```

Comment:
Could you please add a Clear() method to the LoadedPathSegment, call it from the ctor and call it before initialization in factory methods?